doc/datasets.rst

```diff
@@ -0,0 +1,37 @@
+.. _datasets:
+
+========
+Datasets
+========
+
+.. currentmodule:: skrub
+
+skrub provides fetching methods for datasets, 
+which are generally used in examples and benchmarks.
+
+It consists of generated, embeddings and real world data.
+
+Real world datasets
```

Comment:
It looks to me like we are duplicating the API page. I would rather that we point to it rather than establish a second list, as this list will be hard to keep up to date.

Reply:
But why do that? It adds complexity to the API page.

I would rather we don't change the API page, and just do a very small page on the datasets in the narrative docs that points to the API page. The datasets are not a core user-facing aspect of skrub. They are there to enable us to do good examples. They don't need to be prominent in the doc

Reply:
The idea was to list all functions in both the API and datasets file while not duplicating them explicitly.

I understand the complexity it adds and that duplicating is not the best option. 
Here is a very simple alternative version where I only provide a link to the API page.